src/lib/Browser/Element/Condition/ElementHasTextCondition.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Behat\Browser\Element\Condition;
+
+use Ibexa\Behat\Browser\Element\BaseElementInterface;
+use Ibexa\Behat\Browser\Element\Mapper\ElementTextMapper;
+use Ibexa\Behat\Browser\Locator\LocatorInterface;
+
+class ElementHasTextCondition implements ConditionInterface
+{
+    /**
+     * @var \Ibexa\Behat\Browser\Element\BaseElementInterface
+     */
+    private $searchedNode;
+
+    /**
+     * @var \Ibexa\Behat\Browser\Locator\LocatorInterface
+     */
+    private $elementLocator;
+
+    /**
+     * @var string
+     */
+    private $expectedText;
+
+    private $foundElementText;
```

Comment:
```suggestion
    /**
     * @var string[]
     */
    private $foundElementsText;
```

We use `findAll` and potentially find multiple elements matching our selector, so I'd change the variable to plural somehow